mitodl/ocw-studio
# make sure menu items never have zero weight

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-studio/issues/495

#### What's this PR do?
In https://github.com/mitodl/ocw-studio/pull/364, we added the back end support for managing menus.  When this was done, it was set up so the weighting of these items starts at zero.  Hugo does not support weights of zero, so in this PR the code has been altered so the weight starts at 10 and increments from there.  A migration was also added to adjust existing menu items up by 10 for sites created through `ocw-studio`.

#### How should this be manually tested?
 - Spin up a local instance of `ocw-studio` and ensure that Github integration is enabled and pointing at a test organization you've created as detailed here: https://github.com/mitodl/ocw-studio#enabling-github-integration
 - Create a site at http://localhost:8043/sites using a starter that has a navmenu collection, such as [`ocw-course`](https://github.com/mitodl/ocw-hugo-projects/blob/main/ocw-course/ocw-studio.yaml)
 - Go to the Pages section and create some pages in your site
 - Go to the Menu section and add some internal and external link menu items, nesting some of them
 - Save and publish the site to your test Github organization
 - View the published output and browse to `config/_default/menus.yaml` and ensure that no menu items have weights of zero
